lib/streamlit/elements/exception.py

```diff
@@ -145,26 +155,23 @@ def _format_syntax_error_message(exception):
     >                            ^
     > SyntaxError: invalid syntax
 
-    Parameters
-    ----------
-    exception : SyntaxError
-
-    Returns
-    -------
-    str
-
     """
     if exception.text:
+        if exception.offset is not None:
+            caret_indent = " " * max(exception.offset - 1, 0)
+        else:
+            caret_indent = ""
+
         return (
-            'File "%(filename)s", line %(lineno)d\n'
+            'File "%(filename)s", line %(lineno)s\n'
```

Comment:
Changed from `%(lineno)d` to `%(lineno)s` because `lineno` is an `Optional[int]`, and the `d` format specifier will raise an error if its input is not an int.